common/src/main/java/com/microsoft/identity/common/internal/fido/FidoChallengeFactory.kt

```diff
@@ -0,0 +1,112 @@
+         * @throws ClientException if a required parameter is missing.
+         */
+        @JvmStatic
+        @Throws(ClientException::class)
+        fun createFidoChallengeFromRedirect(redirectUri: String): IFidoChallenge {
+            val parameters = UrlUtil.getParameters(redirectUri)
+            //At the moment, only auth FIDO requests will be sent by the server.
+            return AuthFidoChallenge(
+                challenge = validateRequiredParameter(
+                    FidoRequestField.Challenge.name,
+                    parameters[FidoRequestField.Challenge.name]
+                ),
+                relyingPartyIdentifier = validateRequiredParameter(
+                    FidoRequestField.RelyingPartyIdentifier.name,
+                    parameters[FidoRequestField.RelyingPartyIdentifier.name]
+                ),
+                userVerificationPolicy = validateRequiredParameter(
+                    FidoRequestField.UserVerificationPolicy.name,
+                    parameters[FidoRequestField.UserVerificationPolicy.name]
+                ),
+                version = validateRequiredParameter(
+                    FidoRequestField.Version.name,
+                    parameters[FidoRequestField.Version.name]
+                ),
+                submitUrl = validateRequiredParameter(
+                    FidoRequestField.SubmitUrl.name,
+                    parameters[FidoRequestField.SubmitUrl.name]
+                ),
+                keyTypes = validateRequiredListParameter(
+                    FidoRequestField.KeyTypes.name,
+                    parameters[FidoRequestField.KeyTypes.name]
+                ),
+                context = validateRequiredParameter(
+                    FidoRequestField.Context.name,
+                    parameters[FidoRequestField.Context.name]
+                ),
+                allowedCredentials = validateRequiredListParameter(
+                    AuthFidoRequestField.AllowedCredentials.name,
+                    parameters[AuthFidoRequestField.AllowedCredentials.name]
+                )
+            )
+        }
+
+        /**
+         * Validates that the given parameter is not null or empty.
+         * @param field passkey protocol field
+         * @param value value for a passkey protocol parameter
+         * @throws ClientException if the parameter is null or empty.
+         */
+        @Throws(ClientException::class)
+        private fun validateRequiredParameter(field: String, value: String?): String {
+            if (value == null) {
+                throw ClientException(PASSKEY_PROTOCOL_REQUEST_INVALID, "$field not provided")
+            } else if (value.isBlank()) {
+                throw ClientException(PASSKEY_PROTOCOL_REQUEST_INVALID, "$field is empty")
+            }
+            return value;
+        }
+
+        /**
+         * Validates that the given parameter is not null or empty.
+         * @param value list value for a passkey protocol parameter
```

Comment:
```suggestion
         * @param list value for a passkey protocol parameter
```

Reply:
"value" is the parameter name, but I did edit the kotlindocs to be better written,